web/packages/teleterm/src/ui/HeadlessAuthn/HeadlessPrompt/HeadlessPrompt.tsx

```diff
@@ -0,0 +1,127 @@
+ */
+
+import React, { useState } from 'react';
+import * as Alerts from 'design/Alert';
+import { ButtonIcon, Box, Text, ButtonSecondary } from 'design';
+import Dialog from 'design/Dialog';
+import { DialogContent, DialogHeader, DialogFooter } from 'design/Dialog';
+import { PromptWebauthn } from '../../ClusterConnect/ClusterLogin/FormLogin/PromptWebauthn';
+import * as Icons from 'design/Icon';
+
+export type HeadlessPromptProps = {
+  clientIP: string;
+  onApprove(): void;
+  onReject(): void;
+  updateHeadlessStateAttempt: any;
+  onClose(): void;
+  onAbort(): void;
+};
+
+export function HeadlessPrompt({
+  clientIP,
+  onApprove,
+  onReject,
+  updateHeadlessStateAttempt,
+  onClose,
+  onAbort,
+}: HeadlessPromptProps) {
+  const [waitForMfa, setWaitForMfa] = useState(false);
+
+  return (
+    <Dialog
+      dialogCss={() => ({
+        maxWidth: '480px',
+        width: '100%',
+        padding: '0',
+      })}
+      disableEscapeKeyDown={false}
+      open={true}
+    >
+      <Box p={4}>
+        <DialogHeader
+          justifyContent="space-between"
+          mb={0}
+          alignItems="baseline"
+        >
+          <Text typography="h4" bold>
+            Headless Login Detected
+          </Text>
+          <ButtonIcon
+            type="button"
+            onClick={() => {
+              onAbort();
+              onClose();
+            }}
+            color="text.slightlyMuted"
+          >
+            <Icons.Close fontSize={5} />
+          </ButtonIcon>
+        </DialogHeader>
+        <DialogContent>
+          <Text typography="body1" color="text.slightlyMuted">
+            Someone has initiated a command from {clientIP}. If it was not you,
+            click Reject and contact your administrator.
+          </Text>
+        </DialogContent>
+        {updateHeadlessStateAttempt.status === 'error' && (
+          <Alerts.Danger mb={0}>
+            {updateHeadlessStateAttempt.statusText}
+          </Alerts.Danger>
+        )}
+        {!waitForMfa && (
+          <DialogFooter>
+            <ButtonSecondary
+              autoFocus
+              mr={3}
+              type="submit"
+              onClick={async e => {
+                e.preventDefault();
+                setWaitForMfa(true);
+                await onApprove();
+                onClose();
+              }}
+            >
+              Approve
+            </ButtonSecondary>
+            <ButtonSecondary
+              type="button"
+              onClick={async e => {
+                e.preventDefault();
+                await onReject();
+                onClose();
+              }}
+            >
+              Reject
+            </ButtonSecondary>
+          </DialogFooter>
+        )}
+        {waitForMfa && (
+          <DialogContent mb={2}>
+            <Text>Complete MFA verification to approve the Headless Login</Text>
+            <PromptWebauthn
+              prompt={'tap'}
+              onCancel={() => {
```

Comment:
That cancel button looks a little weird in the bottom-right corner IMO as we usually keep them on the left side.
<img width="538" alt="image" src="https://github.com/gravitational/teleport/assets/20583051/c0fd987b-2730-474a-b7c9-0ec9b63f81c7">

But you depend on `PromptWebauthn` here, so I would keep this. 
Later, I or Rafał can work on that.

Reply:
Shouldn't we hide the text "Someone has initiated..." when we are in MFA state? There is no "Reject" button here.